lib/errors/api_error_handling.rb

```diff
@@ -1,14 +1,17 @@
 # frozen_string_literal: true
 
 module ApiErrorHandling
-  def log_error(error_record)
-    Rails.logger.info "Caught #{error_record.error}"
-    error_record.level = error_level(error_record.error)
-    if error_record.specific_error_logging?
-      error_record.log_error
-    else
-      report_exception_sentry(error_record)
+  def handle_api_error(error, update_service: nil, sentry_extra:)
+    Rails.logger.info "Caught #{error}"
+    sentry_tags = nil
+    if update_service.present?
+      update_service.update_error_stats
+      sentry_tags = update_service.sentry_tags
     end
+    Raven.capture_exception(error,
+                            level: error_level(error),
+                            extra: sentry_extra,
+                            tags: sentry_tags)
     return nil
```

Comment:
Looks like this now skips the `raise_unexpected_error`?

Reply:
raise_unexpected_error is not required by Replica so I have called it separately(fid the same way earlier too)

Reply:
Oh, I see... that gets done in the inidividual API classes...

Reply:
I think it makes would makes sense to get rid of that raising behavior altogether. Part of the point here is that we want to make errors easier for end-users to find out about, and if we raise an error and kill the update service process, then the error won't get logged in the course flags and will be harder to find. If we needed to retain the current behavior, where some files can raise an unexpected error but some cannot (eg, `Replica`), I think would make sense to configure whether or not to raise the errors using a configuration constant in each class (`RAISE_UNEXPECTED_ERRORS`). But I think removing that behavior makes more sense.